src/App.tsx

```diff
@@ -15,47 +16,134 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ALPHABET,
+  LENGTH,
+  NONE,
+}
+
+// Use this function in the render to prepare goods
+export function getReorderedGoods(
+  goods: string[],
+  sortType: SortType,
+  isReversed: boolean,
+) {
+  // To avoid the original array mutation
+  const visibleGoods = [...goods];
+
+  // Sort and reverse goods if needed
+  visibleGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+
+      case SortType.NONE:
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  // eslint-disable-next-line no-console
+  console.log(sortType, isReversed);
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setRevers] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const visibleGoods = getReorderedGoods(
+    goodsFromServer,
+    sortType,
+    isReversed,
+  );
+  const isResetNeed = isReversed === true || sortType !== SortType.NONE;
+
+  const sortAlphabeticaly = () => {
+    setSortType(SortType.ALPHABET);
+  };
+
+  const sortByLength = () => {
+    setSortType(SortType.LENGTH);
+  };
+
+  const reverse = () => {
+    setRevers(!isReversed);
+  };
+
+  const reset = () => {
+    setSortType(SortType.NONE);
+    setRevers(false);
+  };
+
```

Comment:
I think by writing all these functions separately, you are writing a lot of redundant code. it would be better to put them directly in the onclick

Reply:
We have opposite opinions on this matter))) We will wait for the mentor's decision)

Reply:
both your opinions are valid, there's no single best approach to writing code in programming.
For production code the functions (except reset) are probably too simple to move them from onClick, but we're learning here so that's a good approach too.